### ISSUE TITLE
System.IndexOutOfRangeException

### ISSUE THREAD

**Issue by gpersell** [PowerShell/PSReadLine#766]
Oops, something went wrong.  Please report this bug with the details below.                         
Report on GitHub: https://github.com/lzybkr/PSReadLine/issues/new                                   
-----------------------------------------------------------------------                             
Last 1 Keys:                                                                                        
 c                                                                                                  
                                                                                                    
Exception:                                                                                          
System.IndexOutOfRangeException: Index was outside the bounds of the array.                         
   at Microsoft.PowerShell.PSConsoleReadLine.ReallyRender()                                         
   at Microsoft.PowerShell.PSConsoleReadLine.Insert(Char c)                                         
   at Microsoft.PowerShell.PSConsoleReadLine.SelfInsert(Nullable`1 key, Object arg)                 
   at Microsoft.PowerShell.PSConsoleReadLine.ProcessOneKey(ConsoleKeyInfo key, Dictionary`2 dispatch
Table, Boolean ignoreIfNoAction, Object arg)                                                        
   at Microsoft.PowerShell.PSConsoleReadLine.InputLoop()                                            
   at Microsoft.PowerShell.PSConsoleReadLine.ReadLine(Runspace runspace, EngineIntrinsics engineIntr
insics)


Environment data
----------------

<!-- provide the output of the following:
```powershell
& {
    "PS version: $($PSVersionTable.PSVersion)"
    $v = (Get-Module PSReadline).Version
    $m = Get-Content "$(Split-Path -Parent (Get-Module PSReadLine).Path)\PSReadLine.psd1" | Select-String "Prerelease = '(.*)'"
    if ($m) {
        $v = "$v-" + $m.Matches[0].Groups[1].Value
    }
    "PSReadline version: $v"
    if ($IsLinux -or $IsMacOS) {
        "os: $(uname -a)"
    } else {
        "os: $((dir $env:SystemRoot\System32\cmd.exe).VersionInfo.FileVersion)"
    }
    "PS file version: $((dir $pshome\p*[hl].exe).VersionInfo.FileVersion)"
}
```
-->

Here is the output of that command:
--------------------------------------
At line:1 char:12
+ powershell & {     "PS version: $($PSVersionTable.PSVersion)"     $v  ...
+            ~
The ampersand (&) character is not allowed. The & operator is reserved for future use; wrap an
ampersand in double quotation marks ("&") to pass it as part of a string.
At line:1 char:67
+ ... ell & {     "PS version: $($PSVersionTable.PSVersion)"     $v = (Get- ...
+                                                                ~~
Unexpected token '$v' in expression or statement.
At line:1 char:108
+ ... e.PSVersion)"     $v = (Get-Module PSReadline).Version     $m = Get-C ...
+                                                                ~~
Unexpected token '$m' in expression or statement.
    + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : AmpersandNotAllowed


Steps to reproduce or exception report
--------------------------------------
I simply opened a powershell shell using cmder and waited.

**Comment by lzybkr** [comment]
It's a little confusing with the markdown and PowerShell in the issue template, but you were supposed to run the command from PowerShell, not as part of a command line that begins with PowerShell.

If you are using PSReadLine 1.2 and don't have reliable repro steps, I think I'll just close this as the rendering code was rewritten in [2.0.0](https://www.powershellgallery.com/packages/PSReadLine/2.0.0-beta3).

**Comment by lzybkr** [comment]
Closing as fixed assuming this was reported against 1.2, please comment and add repro steps if this happened with 2.0.

**Comment by nnrj** [comment]
Setting PowerShell as an default terminal in VScode, and inputing any characters successive. 
Then pressing Backspace successive, some characters near the ">" can not be deleted.
And in some case, the problem will lead PowerShell to crash and print exception information.
The EXCEPTION INFOMATION in a crash matter as follow:
（Noticeable: Anti-virus software is "Total investigation and killing" when the exception happen. )
  (And the words after "//" is single line comment, they are not exception contexts.)

> 上 200 个密钥: //200 secret keys:(The "上" may be an overflow character. )
>  i r t u a Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Enter
>  m v Space v i r t u a n l . h Space v i r t u a l . h Enter
>  c o u d e Space v i r t u a l . h Enter
>  c o u d e Space v i r t u a l . h Enter
>  c o u d e Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace c o u d e Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace c o d e Backspace Backspace Backspace Backspace Backspace c o d e Space v i r t u a l . h Enter
>  g + + Space c Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace d o
> 
> 异常: //Exception:
> System.ArgumentOutOfRangeException: 该值必须大于或等于零，且必须小于控制台缓冲区在该维度的大小。//The value must grater or equal to zero, and it must lower than the size of console's buffer area in this dimension.
> 参数名: left  //Parameter name:left
> 实际值是 -1。//Actual value:-1
>    在 System.Console.SetCursorPosition(Int32 left, Int32 top) //"在" = "in", similarly hereinafter.
>    在 Microsoft.PowerShell.Internal.VirtualTerminal.set_CursorLeft(Int32 value)
>    在 Microsoft.PowerShell.PSConsoleReadLine.ReallyRender(RenderData renderData, String defaultColor)
>    在 Microsoft.PowerShell.PSConsoleReadLine.ForceRender()
>    在 Microsoft.PowerShell.PSConsoleReadLine.Insert(Char c)
>    在 Microsoft.PowerShell.PSConsoleReadLine.SelfInsert(Nullable`1 key, Object arg)
>    在 Microsoft.PowerShell.PSConsoleReadLine.ProcessOneKey(ConsoleKeyInfo key, Dictionary`2 dispatchTable, Boolean ignoreIfNoAction, Object arg)
>    在 Microsoft.PowerShell.PSConsoleReadLine.InputLoop()
>    在 Microsoft.PowerShell.PSConsoleReadLine.ReadLine(Runspace runspace, EngineIntrinsics engineIntrinsics)
